### ISSUE TITLE
Preserve exif data

### ISSUE THREAD

**Issue by claviska** [claviska/SimpleImage#157]
Because of the way GD works, exif data is discarded when an image is saved. It would be really nice to write this data back to the file when saved as `image/jpeg`.

On load, `$this->exif` gets populated from [`exif_read_data()`](http://php.net/manual/en/function.exif-read-data.php), so we have the info. We just need a way to write it back to the JPEG after it gets generated.

There are some seemingly bloated libraries for reading/writing exif data. I'd like to find a lightweight solution to handle this (prefer no third party deps).

- Add `setExif()` method (in case you want to overwrite or change it).
- Add `removeExif()` method (in case you want to remove it from the image).
- Ensure width, height, orientation, etc. exif properties are updated automatically when the image is resized or rotated.

**Comment by karsasmus** [comment]
Hi Cory, perhaps this [Stackoverflow answer](http://stackoverflow.com/a/17890518) could help a little bit

**Comment by claviska** [comment]
Yep, looks like [this comment](http://php.net/manual/en/function.iptcembed.php#113877) might be a good starting point. Thanks!

**Comment by johnabela** [comment]
Would absolutely love to see this get implemented - probably my #1 feature request for SimpleImage.

**Comment by johnabela** [comment]
Been almost a year since this was last discussed, any update on getting this incorporated @claviska ?

**Comment by johnabela** [comment]
Sounds good mate, totally understand. Currently rebuilding a good part of my own base code from the ground up. I will try to invest some time to seeing if I can get some code integrated into SimpleImage that deals with getting this to work. I think my main thing is just getting autoOrient() to work and not have the exif data go bye-bye. It is nice to just run autoOrient() on new uploads just to make sure they are not all wonky, but it sucks having the geo and camera exif killed off, just to perform that simple operation.
